universe.py

```diff
@@ -45,30 +49,47 @@ def main_character(self):
     """
 
     def update(self, events):
+        for the_projectile in self.created_friendly_projectiles:
+            self.friendly_projectiles_[the_projectile.ID_] = the_projectile
+            self.gameobjects_[the_projectile.ID_] = the_projectile
+        
+        for the_enemy in self.created_enemies:
+            self.enemies_[the_enemy.ID_] = the_enemy
+            self.gameobjects_[the_enemy.ID_] = the_enemy
+
+        for the_projectile in self.created_enemy_projectiles: 
+            self.enemy_projectiles_[the_projectile.ID_] = the_projectile
+            self.gameobjects_[the_projectile.ID_] = the_projectile
+
+        for the_main_character in self.created_main_character:
+            self.main_character_ = the_main_character
+            self.gameobjects_[the_main_character.ID_] = the_main_character
+
+        self.created_friendly_projectiles.clear()
+        self.created_enemies.clear()
+        self.created_main_character.clear()
+        self.created_enemy_projectiles.clear()
+
         self.update_collisions()
         self.delete_out_of_bounds_gameobjects()
         for gameobject in self.gameobjects():
             gameobject.update(events)
 
-        print("Collisions : " + str(self.collisions_))
+        #print("Collisions : " + str(self.collisions_))
         # print(str(len(self.gameobjects_)) + " gameobjects")
         # print(str(len(self.enemies_)) + " enemies")
 
     def create_enemy(self, the_enemy):
-        self.enemies_[the_enemy.ID_] = the_enemy
-        self.gameobjects_[the_enemy.ID_] = the_enemy
-
+        self.created_enemies.append(the_enemy)
+        
     def create_enemy_projectile(self, the_projectile):
```

Comment:
Don't think it'd hurt to be more specific here, having the parameter be `the_enemy_projectile` instead of just `the_projectile`